src/commands/gate/utils.ts

```diff
@@ -5,3 +5,31 @@ export const getBaseIntakeUrl = () => {
 
   return 'https://quality-gates.datadoghq.com'
 }
+
+/**
+ * Receives an array of the form ['key:value', 'key2:value2_1', 'key2:value2_2']
+ * and returns an object of the form {key: ['value'], key2: ['value2_1, value2_2']}
+ */
+export const parseScope = (scope: string[]) => {
+  try {
+    return scope.reduce((acc: {[key: string]: string[]}, keyValuePair) => {
+      if (!keyValuePair.includes(':')) {
+        return acc
+      }
+
+      const firstColon = keyValuePair.indexOf(':')
+      const key = keyValuePair.substring(0, firstColon)
+      const value = keyValuePair.substring(firstColon + 1)
+
+      if (acc.hasOwnProperty(key)) {
+        acc[key].push(value)
+      } else {
+        acc[key] = [value]
```

Comment:
❓ do you always want lists? Even for single values? Because there's the option to do: 

```javascript
if (acc.hasOwnProperty(key)) {
  if (acc[key].length) {
    acc[key].push(value)
  } else {
    acc[key] = [acc[key], value]
  }
}
```
it's not the prettiest but maybe it's easier to handle in the backend?

Reply:
I think we are fine with lists of single values. In the backend we will have a specific struct assuming a list there and we will iterate over the elements. If there'll be 1 element it will just do a single iteration.